cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -401,15 +401,39 @@ func (csr *ClusterStateRegistry) GetUnregisteredNodes() []UnregisteredNode {
 
 // GetReadinessState gets readiness state for the node
 func GetReadinessState(node *apiv1.Node) (isNodeReady bool, lastTransitionTime time.Time, err error) {
-	for _, condition := range node.Status.Conditions {
-		if condition.Type == apiv1.NodeReady {
-			if condition.Status == apiv1.ConditionTrue {
-				return true, condition.LastTransitionTime.Time, nil
+	canNodeBeReady, readyFound := true, false
```

Comment:
Please merge GetReadinessState and IsNodeReadyAndSchedulable into one method as they are very similar.